app/models/models.py

```diff
@@ -14,85 +11,209 @@
-        self.flowchart_json = kwargs['flowchart_json']
-    
-    def __repr__(self):
-        return F"Flowchart(id={self.id}, description={self.description}, flowchart_file={self.flowchart_file}')"
 
+class User(db.Model):
+    __tablename__ = 'user'
 
-class Job(db.Model):
-    __tablename__ = 'jobs'
+    id = db.Column(db.Integer, primary_key=True)
+    username = db.Column(db.String, unique=True, nullable=False)
+    first_name = db.Column(db.String)
+    last_name = db.Column(db.String)
+    email = db.Column(db.String)
+    password = db.Column(db.String)
+    added = db.Column(db.DateTime, nullable=False, default=datetime.utcnow)
+    status = db.Column(db.String, default='active')
+
+    groups = db.relationship(
+        'Group', secondary=user_group, back_populates='users'
+    )
+    projects = db.relationship(
+        'Project', secondary=user_project, back_populates='users'
+    )
+
+
+class Group(db.Model):
+    __tablename__ = 'group'
 
     id = db.Column(db.Integer, primary_key=True)
-    path = db.Column(db.String, nullable=False, unique=True)
-    flowchart_id = db.Column(db.String, db.ForeignKey("flowcharts.id"))
-    submission_date = db.Column(db.DateTime, nullable=False, default=datetime.utcnow)
-    author = db.Column(db.String, nullable=True)
-    name = db.Column(db.String, nullable=True)
-    notes = db.Column(db.String, nullable=True)
-    status = db.Column(db.String, nullable=False, default="Created")
+    name = db.Column(db.String, unique=True, nullable=False)
+
+    users = db.relationship(
+        'User', secondary=user_group, back_populates='groups'
+    )
+
 
-    flowchart = db.relationship('Flowchart', back_populates='jobs')
+class Flowchart(db.Model):
+    __tablename__ = 'flowchart'
+
+    id = db.Column(db.String(32), nullable=False, primary_key=True)
+    title = db.Column(db.String(100), nullable=True)
+    description = db.Column(db.Text, nullable=True)
+    owner = db.Column(db.Integer, db.ForeignKey('user.id'))
+    group = db.Column(db.Integer, db.ForeignKey('group.id'))
+    path = db.Column(db.String, nullable=False)
+    text = db.Column(db.Text, nullable=False)
+    json = db.Column(db.JSON, nullable=False)
+    created = db.Column(db.DateTime, nullable=False, default=datetime.utcnow)
+    owner_r = db.Column(db.Boolean, nullable=False, default=True)
+    owner_w = db.Column(db.Boolean, nullable=False, default=True)
+    owner_x = db.Column(db.Boolean, nullable=False, default=True)
+    group_r = db.Column(db.Boolean, nullable=False, default=True)
+    group_w = db.Column(db.Boolean, nullable=False, default=False)
+    group_x = db.Column(db.Boolean, nullable=False, default=True)
+    world_r = db.Column(db.Boolean, nullable=False, default=False)
+    world_w = db.Column(db.Boolean, nullable=False, default=False)
+    world_x = db.Column(db.Boolean, nullable=False, default=False)
+
+    job = db.relationship('Job', back_populates='flowchart', lazy=True)
```

Comment:
Small detail, but I think this should be `jobs` rather than `job` because there could be many jobs associated with a flowchart.

Reply:
Correct & fixed. Also used job.id, ie. the unique number, for the link since jobs may not have titles.